src/test/java/bio/terra/app/usermetrics/BardClientTest.java

```diff
@@ -0,0 +1,139 @@
+package bio.terra.app.usermetrics;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import bio.terra.app.configuration.UserMetricsConfiguration;
+import bio.terra.common.category.Unit;
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.web.client.RestTemplate;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(properties = {"datarepo.testWithEmbeddedDatabase=false"})
+@AutoConfigureMockMvc
+@ActiveProfiles({"google", "unittest"})
+@Category(Unit.class)
+@SuppressWarnings("rawtypes")
+public class BardClientTest {
+
+  @SpyBean UserMetricsConfiguration userMetricsConfiguration;
+
+  private AuthenticatedUserRequest user1;
+
+  private AuthenticatedUserRequest user2;
+
+  private String SYNC_PATH;
+
+  private String API_PATH;
+
+  @Before
+  public void setup() {
+    SYNC_PATH = userMetricsConfiguration.getBardBasePath() + "/api/syncProfile";
+    API_PATH = userMetricsConfiguration.getBardBasePath() + "/api/event";
```

Comment:
Suggestion: could we instead set these in `BardClient` for reuse there, and expose with `@VisibleForTesting`?

Reply:
Should I go a step further and define the paths in the configuration files and expose them through the UserMetricsConfiguration?

Reply:
No, I wouldn't do that: the base path is the only component that could change across environments, whereas the endpoints themselves will remain constant.